core/src/main/java/org/apache/iceberg/rest/requests/UpdateNamespacePropertiesRequest.java

```diff
@@ -0,0 +1,161 @@
+ * The request body specifies a list of properties to remove and a map
+ * of key value pairs to update.
+ * <p>
+ * Properties that are not in the request are not modified or removed by this call.
+ * Server implementations are not required to support namespace properties.
+ */
+public class UpdateNamespacePropertiesRequest {
+
+  private List<String> removals;
+  private Map<String, String> updates;
+
+  public UpdateNamespacePropertiesRequest() {
+    // Required for Jackson deserialization.
+  }
+
+  private UpdateNamespacePropertiesRequest(List<String> removals, Map<String, String> updates) {
+    this.removals = removals;
+    this.updates = updates;
+    validate();
+  }
+
+  UpdateNamespacePropertiesRequest validate() {
+    ValidationException.check((this.removals != null && !this.removals.isEmpty()) ||
+            (this.updates != null && !this.updates.isEmpty()),
+        "Cannot create a request to update a namespace's properties without specifying any props to update or remove");
+    return this;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public List<String> removals() {
+    return removals != null ? ImmutableList.copyOf(removals) : Collections.emptyList();
+  }
+
+  public Map<String, String> updates() {
+    return updates != null ? ImmutableMap.copyOf(updates) : Collections.emptyMap();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    UpdateNamespacePropertiesRequest that = (UpdateNamespacePropertiesRequest) o;
+    return Objects.equals(removals, that.removals) && Objects.equals(updates, that.updates);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(removals, updates);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("removals", removals)
+        .add("updates", updates)
+        .toString();
+  }
+
+  public static class Builder {
+    private ImmutableSet.Builder<String> removalsBuilder;
+    private ImmutableMap.Builder<String, String> updatesBuilder;
+
+    private Builder() {
+    }
+
+    public Builder remove(String removal) {
+      ensureRemovalsInitialized();
+      removalsBuilder.add(removal);
+      return this;
+    }
+
+    public Builder removeAll(Iterable<String> removals) {
+      ensureRemovalsInitialized();
+      removalsBuilder.addAll(removals);
+      return this;
+    }
+
+    public Builder update(String key, String value) {
+      ensureUpdatesInitialized();
+      updatesBuilder.put(key, value);
+      return this;
+    }
+
+    public Builder updateAll(Map<String, String> updates) {
+      ensureUpdatesInitialized();
+      updatesBuilder.putAll(updates);
+      return this;
+    }
+
+    private void ensureRemovalsInitialized() {
+      if (removalsBuilder == null) {
+        removalsBuilder = ImmutableSet.builder();
+      }
+    }
```

Comment:
Lazily instantiated so that we serialize null vs empty in a way that will match what is parsed with equals. Accessor methods return immutable copies, so null values are tolerated.